commit df66ef67c3344b767b3d1d8396d4a3e81942c4e9
Author: Jens Axboe <axboe@kernel.dk>
Date:   Sat Nov 17 07:43:42 2018 -0700

    aio: fix failure to put the file pointer
    
    [ Upstream commit 53fffe29a9e664a999dd3787e4428da8c30533e0 ]
    
    If the ioprio capability check fails, we return without putting
    the file pointer.
    
    Fixes: d9a08a9e616b ("fs: Add aio iopriority support")
    Signed-off-by: Jens Axboe <axboe@kernel.dk>
    Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/fs/aio.c b/fs/aio.c
index b9350f3360c6..04c4d6218978 100644
--- a/fs/aio.c
+++ b/fs/aio.c
@@ -1436,6 +1436,7 @@ static int aio_prep_rw(struct kiocb *req, struct iocb *iocb)
 		ret = ioprio_check_cap(iocb->aio_reqprio);
 		if (ret) {
 			pr_debug("aio ioprio check cap error: %d\n", ret);
+			fput(req->ki_filp);
 			return ret;
 		}
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1058_df66ef67c3344b767b3d1d8396d4a3e81942c4e9.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1058_df66ef67c3344b767b3d1d8396d4a3e81942c4e9.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
